src/audio_spectrogram.py

```
@@ -192,6 +185,16 @@ def generate_spectrogram(
         detail = result.stderr.decode(errors="replace").strip() or "no audio was produced"
         raise RuntimeError(f"FFmpeg could not decode audio stream {stream_index}: {detail}")
 
+    try:
+        import librosa
+        import librosa.display
+        import matplotlib
+        matplotlib.use("Agg")
+        import matplotlib.pyplot as plt
+        import numpy as np
+    except ImportError:
+        raise RuntimeError("Missing required libraries for spectrogram generation. Install librosa and matplotlib.")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)